aptos-move/framework/move-stdlib/sources/cmp.move

```diff
@@ -0,0 +1,133 @@
+module std::cmp {
+    enum Ordering has copy, drop {
+        /// First value is less than the second value.
+        Less,
+        /// First value is equal to the second value.
+        Equal,
+        /// First value is greater than the second value.
+        Greater,
+    }
+
+    // TODO - functions here are `public(friend)` here for one release,
+    // and to be changed to `public` one release later.
+    #[test_only]
+    friend std::bcs_tests;
+
+    /// Compares two values with the natural ordering:
+    /// - native types are compared identically to `<` and other operators
+    /// - complex types
+    ///   - Structs and vectors - are compared lexicographically - first field/element is compared first,
+    ///     and if equal we proceed to the next.
+    ///   - enum's are compared first by their variant, and if equal - they are compared as structs are.
+    native public(friend) fun compare<T>(first: &T, second: &T): Ordering;
+
+    public fun is_eq(self: &Ordering): bool {
```

Comment:
why not just `eq`, `lt`?

Reply:
primarily to align with Rust naming - https://doc.rust-lang.org/std/cmp/enum.Ordering.html

but secondly, eq soudns much more like a binary operator (i.e. `a.eq(b)`), vs is_eq sounds like unary - `compare(a, b).is_eq()`